pkg/drawio/treeNodeInterface.go

```diff
@@ -86,62 +88,77 @@ func setGeometry(tn TreeNodeInterface) {
 }
 
 // /////////////////////////////////////////////////////////////////////
-// getAllNodes() - return all the nodes in the sub tree
-func getAllNodes(tn TreeNodeInterface) []TreeNodeInterface {
-	childrenSet := map[TreeNodeInterface]bool{}
-	squares, icons, lines := tn.children()
-	for _, s := range squares {
-		childrenSet[s] = true
-	}
-	for _, i := range icons {
-		childrenSet[i] = true
+// getSubTreeNodes() - return all the nodes in the sub tree
+
+func getSubTreeNodes(tn TreeNodeInterface) (
+	squares []SquareTreeNodeInterface,
+	icons []IconTreeNodeInterface,
+	lines []LineTreeNodeInterface) {
+	squares, icons, lines = tn.children()
+	for _, child := range joinTnsLists(squares, icons, lines) {
```

Comment:
```suggestion
	for _, child := range squares {
```
Icons and lines have no children. Better save recursive calls

Reply:
icons can have lines as children (we want the line to be attached to routers if needed).
lines currently does not have children, but to be on the safe side, for further implementation, I prefer to assume that lines can have children

Reply:
yes, some are leaves.
but:
the return values for getSubTreeNodes() should contain the node itself. (if not, calling with root node will miss the root).
so, we must call getSubTreeNodes() for the leaves also.
however, a call with a leave will be thin.
i can add a stopping condition for leaves inside getSubTreeNodes(), but I dont think it will save a lot